### ISSUE TITLE
JSX does not mark `React` as used when imported via `import React = require("react");`

### ISSUE THREAD

**Issue by cezary-biernacki** [palantir/tslint#893]
I have react option enabled for "no-unused-variable"

```
 "no-unused-variable": [true, "react"],
```

Still React is reported as unused when imported like this:

```
 import React = require("react);
```

but this import statement works

```
 import * as React from "react";    
```

Example

```
/// <reference path="typings.d.ts" />
import React = require("react"); // TSLint reports "unused variable: 'React'"
const result: JSX.Element = <div>Test</div>;
export = result;
```

It seems that the problem is caused by react-specific import detection code is only in visitNamespaceImport() method (noUnusedVariableRule.ts file), but not in visitImportEqualsDeclaration() method.

**Comment by JKillian** [comment]
Good catch @cezary-biernacki, thanks! Is using `import * as React from "react";` an acceptable workaround for you until this gets fixed?

**Comment by myitcv** [comment]
@cezary-biernacki @JKillian - [according to the spec](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#1133-import-require-declarations) the two are equivalent. So whilst the rule should be fixed for completeness, @cezary-biernacki you should be safe to use the `import` variation instead of the `require(..)` variation.

**Comment by cezary-biernacki** [comment]
@JKillian: Yes, the workaround is acceptable, just it is confusing. I had to debug TSLint itself to discover what kind of syntax works, as the documentation does not provide necessary details.

**Comment by andersekdahl** [comment]
Another variant of this: TSLint does not mark `React` as used if you import it like this:
`import React from 'react'`
The d.ts file from DefinitelyTyped does not contain a default export, but it's perfectly fine to import React like that if you use Babel (since Babel has interop between ES6 modules and CJS modules).

**Comment by adidahiya** [comment]
> Babel has interop between ES6 modules and CJS modules

well it's a bit more than that; Babel automatically creates synthetic default exports. but I agree that the `import React from 'react'` syntax should also be supported along with `import React = require("react")`. as indicated by the labels, this is an easy fix that anyone is welcome to contribute.

**Comment by FlorianTopf** [comment]
There is another issue related to the original problem #689 and it affects imports in jest unit-test cases. The linter also doesn't mark react as used when calling `TestUtils.renderIntoDocument`. I am using tslint 3.8.1.

```JSX
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as TestUtils from 'react-addons-test-utils';

describe('Input', () => {
    it('just an input test', () => {
       const input = TestUtils.renderIntoDocument(
            <input type = 'text' onKeyPress = {keyPressAction.bind(this)}/>
        );
        const inputNode: Element = ReactDOM.findDOMNode(input);
        inputNode.setAttribute('value', 'test');
        expect(input.getAttribute('value')).toEqual('test');
    });
});
```

**Comment by marcusnielsen** [comment]
We are porting from eslint + babel and have the following use case:

* We do all imports in index.js files. 
* We dependency inject all imports into the other files as named object keys.

So in our render.js file we will have the following snippet:

```
export default function create({
  React,
  StyleSheet,
  css,
  styles
}) {
// snip with tsx stuff
}
```

But setting `"no-unused-variable": [true, "react"]` is not working.

**Comment by adidahiya** [comment]
at this point I would recommend switching to the compiler options and disabling no-unused-variable, see https://github.com/palantir/tslint/issues/1481 and linked threads.

**Comment by andy-hanson** [comment]
Should be fixed by #2235, which uses TypeScript's implementation for `no-unused-variable`.
